src/webapp/features/rabble/Buttons.tsx

```diff
@@ -58,37 +60,94 @@ const Buttons = ({
     }
   }, [currentPlay.played, endTurn, cleanUp]);
 
+  // handlers for gameplay buttons
+  const handlePass = () => {
+    const confirmed = window.confirm(
+      "Are you sure you want to Pass your turn?"
+    );
+    if (confirmed) {
+      endTurn();
+    }
+  };
+
+  const handleSwap = () => {
+    dispatch(clearPlayTiles());
+    dispatch(changeSquareSelection(null));
+    setShowSwapModal(true);
+  };
+
+  const handleShuffle = () => {
+    shuffleTiles(tileRack);
+    reorderRackTiles(tileRack);
+    dispatch(clearPlayTiles());
+    dispatch(changeSquareSelection(null));
+  };
+
+  const handleUndo = () => {
+    dispatch(clearPlayTiles());
+    dispatch(changeSquareSelection(null));
+    dispatch(updateRackTiles(tileRack));
+    cleanUp();
+  };
+
+  const handlePlay = () => {
+    if (playSquares.length) {
+      playWord(playSquares);
+      dispatch(clearPlayTiles());
+      dispatch(changeSquareSelection(null));
+    }
+  };
+
   return (
     <div className={styles.buttons}>
       {currentPlayerHasTurn && (
-        <button
-          className={styles.pass}
-          onClick={() => {
-            const confirmed = window.confirm(
-              "Are you sure you want to Pass your turn?"
-            );
-            if (confirmed) {
-              endTurn();
-            }
-          }}
-        >
-          <FlagIcon />
-          PASS
-        </button>
+        <Button
+          btnclass="pass"
+          onClick={handlePass}
+          content="PASS"
+          children={<FlagIcon />}
+        />
+        // <button
+        //   className={styles.pass}
+        //   onClick={() => {
+        //     const confirmed = window.confirm(
+        //       "Are you sure you want to Pass your turn?"
+        //     );
+        //     if (confirmed) {
+        //       endTurn();
+        //     }
+        //   }}
+        // >
+        //   <FlagIcon />
+        //   PASS
+        // </button>
```

Comment:
Ok, in the commit that cleans up your work-zone comments, let's also get rid of these previous button onclick implementations that are commented out.

Nice refactor!

Reply:
done! I just wanted to give you a chance to review them side-by-side if you thought I needed changes.